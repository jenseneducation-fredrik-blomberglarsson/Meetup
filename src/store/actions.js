import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  }
}

export default actions;