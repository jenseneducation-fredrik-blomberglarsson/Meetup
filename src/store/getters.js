const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getPreviousMeetups: state => {
    return state.previousMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  },
  getBurgerToggle: state => {
    return state.burgerToggle;
  }
}


export default getters;