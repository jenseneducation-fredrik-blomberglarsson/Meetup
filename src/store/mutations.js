const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;