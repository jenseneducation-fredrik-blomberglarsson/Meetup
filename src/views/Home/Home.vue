<template>
  <div class="home">
    <hr class="hr" />
    <section class="heroContainer">
      <h1 class="descHeader">Meetups of every kind</h1>
      <router-link to="/meetups">
        <button class="joinMeetupBtn">Join Meetup</button>
      </router-link>
    </section>
    <section class="informationContainer">
      <p class="infoParagraph">
        Meetups is a platform for finding and building local communities.
        <br />People use Meetups to meet new people,
        <br />learn new things, find support,
        <br />get out of their comfort zones,
        <br />and pursue their passions, together.
      </p>
    </section>
  </div>
</template>

<script>
// @ is an alias to /src
//import HelloWorld from '@/components/HelloWorld.vue'

export default {
  name: "Home",
  components: {},
};
</script>

<style scoped>
@import url("https://fonts.googleapis.com/css2?family=Permanent+Marker&display=swap");
.home {
  width: 100%;
  height: 120vh;
}
.hr {
  width: 100%;
  opacity: 0.5;
}
.heroContainer {
  margin-top: 4%;
  background-image: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)),
    url("../../assets/meetup-home.jpeg");
  height: 500px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  justify-content: center;
}
.descHeader {
  font-size: 2.5rem;
  text-transform: uppercase;
  letter-spacing: 2px;
  border: 3px solid #fff;
  margin-top: 5%;
  align-self: center;
  color: #fff;
  text-align: center;
}
.joinMeetupBtn {
  font-family: "Permanent Marker", cursive;
  color: #fff;
  position: absolute;
  top: 58%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 200px;
  height: 60px;
  text-align: center;
  line-height: 60px;
  color: #fff;
  font-size: 1rem;
  text-transform: uppercase;
  text-decoration: none;
  letter-spacing: 2px;
  box-sizing: border-box;
  background: linear-gradient(90deg, #03a9f4, #f441a5, #ffeb3b, #03a9f4);
  background-size: 400%;
  z-index: 1;
  cursor: pointer;
}
.joinMeetupBtn:hover {
  animation: animate 8s linear infinite;
}
@keyframes animate {
  0% {
    background-position: 0%;
  }
  100% {
    background-position: 400%;
  }
}
.joinMeetupBtn:before {
  content: "";
  position: absolute;
  top: -5px;
  left: -5px;
  right: -5px;
  bottom: -5px;
  z-index: -1;
  background: linear-gradient(90deg, #03a9f4, #f441a5, #ffeb3b, #03a9f4);
  background-size: 400%;
  border-radius: 40px;
  opacity: 0;
  transition: 0.5s;
}
.joinMeetupBtn:hover:before {
  filter: blur(20px);
  opacity: 1;
  animation: animate 8s linear infinite;
}
.infoParagraph {
  margin-top: 5%;
  font-size: 1.5rem;
}
</style>
