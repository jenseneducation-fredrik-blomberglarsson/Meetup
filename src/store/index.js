import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});