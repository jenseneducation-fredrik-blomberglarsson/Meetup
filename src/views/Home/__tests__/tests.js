import { shallowMount, mount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import Home from '@/views/Home/Home.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('The page is rendered', () => {
  test('Test so that everything renders properly', async () => {
    const wrapper = shallowMount(Home, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  })

  test('test if router-link for meetups exists in Home view', () => {
    const wrapper = mount(Home, {
      store,
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.findComponent(RouterLinkStub).props().to).toBe('/meetups');
  });
})