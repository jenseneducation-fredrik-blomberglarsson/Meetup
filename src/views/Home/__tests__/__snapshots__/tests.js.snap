// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page is rendered Test so that everything renders properly 1`] = `
<div
  class="home"
>
  <hr
    class="hr"
  />
   
  <section
    class="heroContainer"
  >
    <h1
      class="descHeader"
    >
      Meetups of every kind
    </h1>
     
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/meetups"
    >
      <button
        class="joinMeetupBtn"
      >
        Join Meetup
      </button>
    </router-link-stub>
  </section>
   
  <section
    class="informationContainer"
  >
    <p
      class="infoParagraph"
    >
      
      Meetups is a platform for finding and building local communities.
      
      <br />
      People use Meetups to meet new people,
      
      <br />
      learn new things, find support,
      
      <br />
      get out of their comfort zones,
      
      <br />
      and pursue their passions, together.
    
    </p>
  </section>
</div>
`;
